src/System.Private.Xml/src/System/Xml/Serialization/XmlReflectionImporter.cs

```diff
@@ -420,7 +420,7 @@ private TypeMapping ImportTypeMapping(TypeModel model, string ns, ImportContext
                     case TypeKind.Root:
                     case TypeKind.Class:
                     case TypeKind.Struct:
-                        if (context != ImportContext.Element) throw UnsupportedException(model.TypeDesc, context);
+                        if (context != ImportContext.Element && !model.TypeDesc.IsOptionalValue) throw UnsupportedException(model.TypeDesc, context);
                         if (model.TypeDesc.IsOptionalValue)
```

Comment:
Previously this line semantically meant if this is Text or Attribute, throw. While this change is intended for attributes, it also affects Text. Is this okay? Or do we want to narrow down the constraint to:   
`context == ImportContext.Text || (context == ImportContext.Attribute && !model.TypeDesc.IsOptionalValue)`  
That's some difficult logic to conceptualize so it might be better to move the exception into the if and else statements. So:
``csharp
if (model.TypeDesc.IsOptionalValue)
{
    if (context == ImportContext.Text) throw.....
    ....
}
else
{
    if (context != ImportContext.Element) throw.....
    ....
}
```

While it's a bigger code change, that feels easier to read to me.